src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -1438,6 +1470,16 @@ void GcsActorManager::Initialize(const GcsInitData &gcs_init_data) {
     }
   }
   if (!dead_actors.empty()) {
+    for (const auto &actor_id : dead_actors) {
+      auto *actor = GetActor(actor_id);
+      RAY_CHECK(actor != nullptr);
+      RAY_LOG(INFO).WithField(kLogKeyActorID, actor_id)
+          << "Destroying actor because it should be dead when GCS restarts.";
+      if (actor->GetState() != rpc::ActorTableData::DEAD) {
+        // destroy the actor
+        DestroyActor(actor_id, GenActorOutOfScopeCause(actor));
```

Comment:
Will this work? the actor_id is not added to the `registered_acotrs_` so DestroyActor will just early return?

Reply:
fixed